src/java.naming/share/classes/com/sun/jndi/ldap/AbstractLdapNamingEnumeration.java

```diff
@@ -364,49 +448,41 @@ protected final boolean hasMoreReferrals() throws NamingException {
      * with those of the current enumeration.
      */
     protected void update(AbstractLdapNamingEnumeration<? extends NameClassPair> ne) {
-        // Cleanup previous context first
-        homeCtx.decEnumCount();
-
-        // New enum will have already incremented enum count and recorded clnt
-        homeCtx = ne.homeCtx;
-        enumClnt = ne.enumClnt;
-
-        // Do this to prevent referral enumeration (ne) from decrementing
-        // enum count because we'll be doing that here from this
-        // enumeration.
-        ne.homeCtx = null;
-
-        // Record rest of information from new enum
-        posn = ne.posn;
-        limit = ne.limit;
-        res = ne.res;
-        entries = ne.entries;
-        refEx = ne.refEx;
-        listArg = ne.listArg;
-        // record a previous exception and quit if any limit is reached
-        if (errEx == null || ne.errEx instanceof LimitExceededException) {
-            errEx = ne.errEx;
+        try {
+            // Cleanup previous context first
+            getHomeCtx().decEnumCount();
```

Comment:
What is the reason for calling `getHomeCtx()` here when we use `enumCtxt.homeCtx` a few lines below?